File: core/src/main/java/mifos/org/mobilewallet/core/data/fineract/api/services/RegistrationService.java
package mifos.org.mobilewallet.core.data.fineract.api.services;

import mifos.org.mobilewallet.core.data.fineract.api.ApiEndPoints;
import mifos.org.mobilewallet.core.data.fineract.entity.register.RegisterPayload;
import mifos.org.mobilewallet.core.data.fineract.entity.register.UserVerify;
import okhttp3.ResponseBody;
import retrofit2.http.Body;
import retrofit2.http.POST;
import rx.Observable;

public interface RegistrationService {

    @POST(ApiEndPoints.REGISTRATION)
    Observable<ResponseBody> registerUser(@Body RegisterPayload registerPayload);

    @POST(ApiEndPoints.REGISTRATION + "/user")
    Observable<ResponseBody> verifyUser(@Body UserVerify userVerify);
}

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/RegistrationService.kt
package org.mifos.mobilewallet.core.data.fineract.api.services

import okhttp3.ResponseBody
import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints
import com.mifos.mobilewallet.model.entity.register.RegisterPayload
import com.mifos.mobilewallet.model.entity.register.UserVerify
import retrofit2.http.Body
import retrofit2.http.POST
import rx.Observable

interface RegistrationService {
    @POST(ApiEndPoints.REGISTRATION)
    fun registerUser(@Body registerPayload: RegisterPayload): Observable<ResponseBody>

    @POST(ApiEndPoints.REGISTRATION + "/user")
    fun verifyUser(@Body userVerify: UserVerify): Observable<ResponseBody>
}